sql/api/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1382,6 +1382,17 @@ class Column(val node: ColumnNode) extends Logging {
    */
   def over(): Column = over(Window.spec)
 
+  /**
+   * TODO: Add documentation
+   * @return
+   */
+  def outer(): Column = withOrigin {
+    node match {
+      case attr: UnresolvedAttribute if !attr.isMetadataColumn =>
+        Column(LazyOuterReference(attr.nameParts, attr.planId))
+      // TODO: other cases
```

Comment:
Is it ok to use metadata columns as outer?
Do we have other cases we can support `outer()`?